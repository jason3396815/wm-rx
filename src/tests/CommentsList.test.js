import { h, Component } from 'preact';
import CommentsList from '../components/CommentsList';

describe('<CommentsList />', () => {
	it('should render a list of comments', () => {
		const comments = [
			{
				level: 0,
				time_ago: '3 days ago',
				user: 'vutran',
				comments: [],
				content: 'Test body',
			},
			{
				level: 0,
				time_ago: '5 days ago',
				user: 'vutran',
				comments: [
					{
						level: 0,
						time_ago: '4 days ago',
						user: 'tranvu',
						comments: [],
						content: 'Test reply body',
					},
				],
				content: 'Test body',
			},
		];
		expect(<CommentsList comments={comments} />).toMatchSnapshot();
	});
});


