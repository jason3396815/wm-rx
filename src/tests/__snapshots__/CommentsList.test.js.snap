// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<CommentsList /> should render a list of comments 1`] = `
e {
  "attributes": Object {
    "comments": Array [
      Object {
        "comments": Array [],
        "content": "Test body",
        "level": 0,
        "time_ago": "3 days ago",
        "user": "vutran",
      },
      Object {
        "comments": Array [
          Object {
            "comments": Array [],
            "content": "Test reply body",
            "level": 0,
            "time_ago": "4 days ago",
            "user": "tranvu",
          },
        ],
        "content": "Test body",
        "level": 0,
        "time_ago": "5 days ago",
        "user": "vutran",
      },
    ],
  },
  "children": Array [],
  "key": undefined,
  "nodeName": [Function],
}
`;
